src/storage.rs

```diff
@@ -68,26 +68,49 @@ impl Default for AccountBasic {
     }
 }
 
-/// EVM Code, currently mapping to REVM's [ByteCode::LegacyAnalyzed].
-// TODO: Support raw legacy & EOF
 #[derive(Debug, Clone, PartialEq, Default, Serialize, Deserialize)]
-pub struct EvmCode {
-    /// Bytecode with 32 zero bytes padding
+pub struct LegacyAnalyzedCode {
+    /// Bytecode with 32 zero bytes padding.
     bytecode: Bytes,
-    /// Original bytes length
+    /// Original bytes length.
     original_len: usize,
     /// Jump table.
     jump_table: Arc<BitVec<u8>>,
 }
 
+#[derive(Debug, Clone, PartialEq, Default, Serialize, Deserialize)]
+pub struct Eip7702Code {
+    /// EOA which which will inherit the bytecode.
+    delegated_address: Address,
+    /// Version of the bytecode.
+    version: u8,
+    /// Raw bytecode encoding MAGIC, version and delegated address.
+    raw: Bytes,
+}
+
+/// EVM Code, currently mapping to REVM's [ByteCode].
+// TODO: Support raw legacy & EOF
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub enum EvmCode {
+    /// Analyzed legacy code, with jump table.
+    LegacyAnalyzed(LegacyAnalyzedCode),
```

Comment:
Just `Legacy` is good as we should map both non & analyzed REVM bytecodes to the same analyzed structure.